app/experimenter/nimbus-ui/src/types/updateExperimentStatus.ts

```diff
@@ -3,22 +3,15 @@
 // @generated
 // This file was automatically generated and should not be edited.
 
-import { ExperimentInput, NimbusExperimentStatus } from "./globalTypes";
+import { ExperimentInput } from "./globalTypes";
 
 // ====================================================
 // GraphQL mutation operation: updateExperimentStatus
 // ====================================================
 
-export interface updateExperimentStatus_updateExperiment_nimbusExperiment {
-  __typename: "NimbusExperimentType";
-  status: NimbusExperimentStatus | null;
-}
-
 export interface updateExperimentStatus_updateExperiment {
   __typename: "UpdateExperiment";
   message: ObjectField | null;
-  status: number | null;
-  nimbusExperiment: updateExperimentStatus_updateExperiment_nimbusExperiment | null;
```

Comment:
I like the combination into a single mutation, but these type files seem to basically do... nothing for us now? They're all:

```
 __typename: "UpdateExperiment";
  message: ObjectField | null;
```

I liked being able to see what I was "allowed" to change per-page, but guess we can always go look at the fields in `UPDATE_EXPERIMENT_MUTATION` and infer it. I know they're auto-generated but it seems like in this case, we don't even need `updateExperimentStatus`, `updateExperimentBranches`, `updateExperimentAudience` etc.?

Reply:
Oh yeah you could just delete those and change their imports to `updateExperiment_updateExperiment`. `generate-types` doesn't seem to regenerate them.

Reply:
Yeah I think it's a combination of not using hte return values and combining everything into a single mutation.  You're 'allowed' to update anything.  I think a lot of this was built around a set of expectations/designs that didn't actually end up being used, so right now it's a lot of unused code.  This doesn't stop us from changing the way it works, but I think that might be easier if it's clear which parts are being used instead of working around code that's built but doesn't actually do anything?

Reply:
Yeah, I guess by "allowed" I meant it was nice to refer to those types as a sort of guide on what I _should_ be updating on that page.

I'm cool with it though, but maybe as @jodyheavener suggested we can remove even more code and just use `updateExperiment_updateExperiment`? 🧁

Reply:
Yeah it's kind of misleading, cuz the explicit fields are fields being READ not UPDATED, so it's not exactly a good reference for what the page is updating.

Reply:
I'm actually a little surprised `generate-types` didn't remove them for us 🤷

Reply:
What we could do is make type declarations for the set of fields each page updates and then type the inputs in the actual components with those?

Reply:
Like we kinda do that here:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx#L52-L60

but we could pull that out into a type declaration somewhere else to say that's the input type for that page form so we have it explicitly declared somewhere that those are the fields updated by that form?

Reply:
That's certainly doable, but isn't that effectively re-adding what we're taking away here, and then adding the onus of us maintaining them in two spots? With the GQL query the types were automatic 😆

Reply:
No that's what I mean, the auto generated ones this is removing show fields being READ, not WRITTEN.  So you could add or remove arbitrary fields to/from that list without affecting the actual behaviour, it's misleading to refer to that as the list of fields a page updates.  Right?

Reply:
Oh sorry I was actually referring to the input types, but it looks like those were yanked even before this PR and are all `ExperimentInput` now. Yeah I see what you're saying, might call it a followup though.

Reply:
@jaredlockhart Yeah I like that idea. It's cool in a follow up to me though (would prefer it be a high priority maintenance item). PR seems good to me. 😎

Reply:
Followup here

https://github.com/mozilla/experimenter/issues/4631

Reply:
Okay I removed those extra result types and updated the references, good catch @jodyheavener